nav/dataset/dataset.py

```
@@ -1,12 +1,60 @@
+    @property
+    def label_file_path(self) -> Path:
+        if self._label_file_path is None:
+            self._label_file_path = cast(Path, self.label_file_url.retrieve())
+        return self._label_file_path
+
+
+@dataclass
+class ImageFiles:
+    image_files: list[ImageFile]
+
+    def __iter__(self):
```

Ruff (0.14.0)

Missing return type annotation for special method `__iter__`

(ANN204)